src/ScottPlot5/ScottPlot5/Fonts.cs

```diff
@@ -35,7 +66,45 @@ public static class Fonts
     /// </summary>
     public static bool Exists(string fontName)
     {
-        return GetInstalledFonts().Contains(fontName);
+        return (FontResolver is { } && FontResolver.Exists(fontName)) || GetInstalledFonts().Contains(fontName);
+    }
+
+    /// <summary>
+    /// Returns a cached typeface (if one exists in the cache) or
+    /// a new instance to a typeface that most closely matches the requested family name and style
+    /// </summary>
+    public static SKTypeface GetTypeface(string fontName, bool bold, bool italic)
+    {
+        int hashCode = GetTypefaceHashCode(fontName, bold, italic);
+        if (TypefaceCache.TryGetValue(hashCode, out var typeface))
+        {
+            return typeface;
+        }
+
+        //Debug.WriteLine($"{hashCode}: {fontName}, {bold}, {italic}");
+
+        if (FontResolver is { })
+        {
+            typeface = FontResolver.CreateTypeface(fontName, bold, italic);
+            if (typeface is { })
+            {
+                TypefaceCache.Add(hashCode, typeface);
+                return typeface;
+            }
+        }
+
+        SKFontStyleWeight weight = bold ? SKFontStyleWeight.Bold : SKFontStyleWeight.Normal;

```

Comment:
I think all of this stuff should just live in a default `IFontResolver`, rather than being the fallback if nothing can be found by font resolver.

Reply:
This has a slightly different meaning. `IFontResolver `should only handle custom fonts. If a standard font is specified (for example: `ScottPlot.Fonts.Sans`, `ScottPlot.Fonts.Serif`, `ScottPlot.Fonts.Monospace`), then we simply pass them to `SKTypeface.FromFamilyName`. `IFontResolver `should not create them.

Reply:
This is for custom fonts is it not? A user can request a font by name (e.g. Arial), and provided that it is installed on their machine, they will get Arial. Even though Arial is (to my knowledge) not a standard font. This isn't for retrieving the Sans Serif font that ScottPlot would use by default, I don't see a reason why that code shouldn't be left as it is.

Reply:
> This has a slightly different meaning

> This is for custom fonts is it not?

These are some good points! ... and an example of where "naming is hard". One could say that a default system font made bold has been "customized", so the language is a bit squishy. I may rename this class to `FontFileResolver` and supply it with ScottPlot, but not use it as the default because the default behavior is to use system fonts.